### ISSUE TITLE
x?.y works even when x is not an optional

### ISSUE THREAD

**Issue by kach** [accordproject/ergo#387]
Because of the way `?.` is implemented, things like this happen:

```
ergo$ return {a: 1}?.a;
Response. none
  :  {emit: Unit[], response: Nothing?, state: Unit}?
```

(This should be a type error, I think, because people will surely try to do this and get confused.)

**Comment by jeromesimeon** [comment]
I agree. I seem to recall this is the semantics in Swift which I started from, but I don't like it.

**Comment by kach** [comment]
Can be compiled to `(let $opt : Any? = <operand> ; match …)`, where the `let` just forces the operand to be of an optional type.

**Comment by jeromesimeon** [comment]
Looking at the example more carefully, this is definitely *not* the semantic in Swift. It's certainly a bug.
